monitoring/prometheus/alerts/test_alerts.yml

```diff
@@ -792,6 +795,731 @@ tests:
            severity: warning
            type: ceph_default
          exp_annotations:
+           documentation: https://docs.ceph.com/en/latest/rados/operations/health-checks/#slow-ops
            description: >
              1 OSD requests are taking too long to process
              (osd_op_complaint_time exceeded)
+
+# CEPHADM orchestrator alert triggers
+ - interval: 30s
+   input_series:
+    - series: 'ceph_health_detail{name="UPGRADE_EXCEPTION"}'
+      values: '1+0x40'
+   promql_expr_test:
+     - expr: ceph_health_detail{name="UPGRADE_EXCEPTION"} > 0
+       eval_time: 2m
+       exp_samples:
+         - labels: '{__name__="ceph_health_detail", name="UPGRADE_EXCEPTION"}'
+           value: 1
+   alert_rule_test:
+    - eval_time: 1m
+      alertname: Cluster upgrade has failed
+    - eval_time: 5m
+      alertname: Cluster upgrade has failed
```

Comment:
Nice improvement! Do you find the `promql_expr_test` useful? I commented with @aaSharma14 that perhaps we could just go for testing the `alert_rule_test`. For the `promql_expr_test` I'd like to find a way to extract those from the Grafana JSON/jsonnet files and feed this with some captured metrics (rather than synthetic ones).